metadata-ingestion/src/datahub/ingestion/source/vertica/common.py

```diff
@@ -0,0 +1,1573 @@
+            customProperties=properties,
+        )
+        dataset_snapshot.aspects.append(dataset_properties)
+
+        # extra_tags = self.get_extra_tags(inspector, schema, table)
+        extra_tags = list()
+        pk_constraints: dict = inspector.get_pk_constraint(projection, schema)
+
+        foreign_keys = self._get_foreign_keys(dataset_urn, inspector, schema, projection)
+
+        schema_fields = self.get_schema_fields(
+            dataset_name, columns, pk_constraints, tags=extra_tags
+        )
+        schema_metadata = get_schema_metadata(
+            self.report,
+            dataset_name,
+            self.platform,
+            columns,
+            pk_constraints,
+            foreign_keys,
+            schema_fields,
+        )
+        dataset_snapshot.aspects.append(schema_metadata)
+        db_name = self.get_db_name(inspector)
+
+        # table_tags = self.get_extra_tags(inspector, schema, table)
+
+        tags_to_add = []
+        if table_tags:
+            tags_to_add.extend(
+                [make_tag_urn(f"{table_tags.get(projection)}")]
+            )
+            yield self.gen_tags_aspect_workunit(dataset_urn, tags_to_add)
+
+        yield from self.add_table_to_schema_container(dataset_urn, db_name, schema)
+        mce = MetadataChangeEvent(proposedSnapshot=dataset_snapshot)
+        wu = SqlWorkUnit(id=dataset_name, mce=mce)
+        self.report.report_workunit(wu)
+        yield wu
+        dpi_aspect = self.get_dataplatform_instance_aspect(dataset_urn=dataset_urn)
+        if dpi_aspect:
+            yield dpi_aspect
+        subtypes_aspect = MetadataWorkUnit(
+            id=f"{dataset_name}-subtypes",
+            mcp=MetadataChangeProposalWrapper(
+                entityType="dataset",
+                changeType=ChangeTypeClass.UPSERT,
+                entityUrn=dataset_urn,
+                aspectName="subTypes",
+                aspect=SubTypesClass(typeNames=["Projections"]),
+            ),
+        )
+        self.report.report_workunit(subtypes_aspect)
+        yield subtypes_aspect
+
+        yield from self._get_domain_wu(
+            dataset_name=dataset_name,
+            entity_urn=dataset_urn,
+            entity_type="dataset",
+            sql_config=sql_config,
+        )
+
+    def _get_projection(
+        self, dataset_name: str, inspector: Inspector, schema: str, projection: str
+    ) -> List[dict]:
+        columns = []
+        try:
+            columns = inspector.get_projection(projection, schema)
+
+            if len(columns) == 0:
+                self.report.report_warning(MISSING_COLUMN_INFO, dataset_name)
+        except Exception as e:
+            self.report.report_warning(
+                dataset_name,
+                f"unable to get column information due to an error -> {e}",
+            )
+        return columns
+
+    def get_extra_tags(
+        self, inspector: Inspector, schema: str, table: str
+    ) -> Optional[Dict[str, str]]:
```

Comment:
please add a comment on what this method does for vertica

Reply:
Will do for sure .